openhtf/util/measurements.py

```diff
@@ -336,14 +340,15 @@ def value(self):
     """The values stored in this record.
 
     Returns:
-      A list of tuples; the last element of each tuple will be the measured
-    value, the other elements will be the assocated coordinates.  The tuples
-    are output in the order in which they were set.
+      A list of dicts mapping coordinates to values measured at those
+      coordinates. Elements are output in the order in which they were set.
     """
     if not self.is_value_set:
       raise MeasurementNotSetError('Measurement not yet set', self.name)
-    return [dimensions + (value,) for dimensions, value in
-            self.value_dict.iteritems()]
+    return [{
```

Comment:
this is a bit of a contrived return type to suit JSON-friendly output, which isn't really part of this interface.  I'd rather move the dict-ification of these tuples at whatever layer is doing the JSON serialization itself.  Notably, dicts have a nontrivial amount of overhead compared to tuples, and you could potentially have a lot of them here.  also, this return value isn't really suitable for any kind of numerical analysis.  you could return namedtuples of `(dimensions, value)` if you want to avoid the weird semantic of "the last element is the value", but it was intended to represent a point in n-dimensional space, in which case the flat tuple makes the most sense.